plugin/system_copy.vim

```diff
@@ -44,12 +50,29 @@ function! s:CopyCommandForCurrentOS()
   endif
 endfunction
 
+function! s:PasteCommandForCurrentOS()
+  let os = substitute(system('uname'), '\n', '', '')
+  if has("gui_mac") || os == 'Darwin'
+    return 'pbpaste'
+  elseif has("gui_win32")
+    return 'paste'
+  else
+    return 'xsel --clipboard --output'
+  endif
+endfunction
```

Comment:
Looks like we have some duplication in the OS detection code. Could you
extract that to a function like `s:CurrentOS()` that returns one of `["mac",
"windows", "linux"]`, then refactor `s:PasteCommandForCurrentOS()` and
`s:CopyCommandForCurrentOS()` to both use that function and reference the
explicit set of OS names?

``` vim
function s:CopyCommandForCurrentOS()
  let current_os = s:CurrentOS()
  if current_os == "mac"
    return "pbpaste"
  elseif current_os == "windows"
    return 'paste'
  elseif current_os == "linux"
    return 'xsel --clipboard --output'
  else
    echoerr "SystemCopy: OS not recognized."
  endif
endfunction
```